src/k8s-extension/azext_k8s_extension/partner_extensions/DataProtectionKubernetes.py

```diff
@@ -128,11 +143,26 @@ def Update(
         if configuration_settings is None:
             configuration_settings = {}
 
+        bsl_specified = False
         if len(configuration_settings) > 0:
             bsl_specified = self.__is_bsl_specified(configuration_settings)
             self.__validate_and_map_config(configuration_settings, validate_bsl=bsl_specified)
             if bsl_specified:
                 self.__validate_backup_storage_account(cmd.cli_ctx, resource_group_name, cluster_name, configuration_settings)
+        # this step is for brownfield migrating to AAD
+        if configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_USE_AAD) is not None and configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_USE_AAD).lower() == "true":
+            logger.warning("useAAD flag is set to true. Please provide extension MSI Storage Blob Data Contributor role to the storage account.")
+
+        # SA details provided in user inputs, but did not provide SA URI.
+        if bsl_specified and configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI) is None:
+            logger.warning("storageAccountURI is not populated. Setting it to the storage account URI of provided storage account")
+            configuration_settings[self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI] = self.__get_storage_account_uri(cmd.cli_ctx, configuration_settings)
+            logger.warning(f"storageAccountURI: {configuration_settings[self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI]}")
+        # SA details not provided in user input, SA Uri not provided in user input, and also not populated in the original extension, we populate it.
+        elif not bsl_specified and original_extension.configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI) is None and configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI) is None:
```

Comment:
If one just provides a BSL Uri for the current BSL, then I think it's not honored anymore.